content/docs/code-splitting.md

```diff
@@ -0,0 +1,205 @@
+[Code-Splitting](https://webpack.js.org/guides/code-splitting/) is a feature
+of bundlers like Webpack and Browserify (via
+[factor-bundle](https://github.com/browserify/factor-bundle)) which can create
+multiple bundles that can be dynamically loaded at runtime.
+
+Code-splitting your app can help you "lazy-load" just the things that are
+currently needed by the user, which can dramatically improve the performance of
+your app. Overall you haven't reduced the amount of code in your app, but
+you've avoided loading code that the user may never need.
+
+## `import()`
+
+The best way to introduce code-splitting into your app is through the dynamic
+`import()` syntax.
+
+**Before:**
+
+```js
+import { add } from './math';
+
+console.log(add(16, 26));
+```
+
+**After:**
+
+```js
+import("./math").then(math => {
+  console.log(math.add(16, 26));
+});
+```
+
+> Note: The dynamic `import()` syntax is a ECMAScript (JavaScript)
+> [proposal](https://github.com/tc39/proposal-dynamic-import) not currently
+> accepted into the language standard. It is expected to be accepted within the
+> near future.
+
+When Webpack comes across this syntax, it automatically start code-splitting
+your app. If you're using Create React App, this is already configured for you
+and you can start using it immediately.
+
+If you're setting up Webpack yourself, you'll probably want to read Webpack's
+[guide on code splitting](https://webpack.js.org/guides/code-splitting/).
+
+Your Webpack config should look vaguely like this:
+
+```js
+module.exports = {
+  entry: {
+    main: './src/app.js',
+  },
+  output: {
+    // `filename` provides a template for naming your bundles (remember to use `[name]`)
+    filename: '[name].bundle.js',
+    // `chunkFilename` provides a template for naming code-split bundles (optional)
+    chunkFilename: '[name].bundle.js',
+    // `path` is the folder where Webpack will place your bundles
+    path: './dist',
+    // `publicPath` is where Webpack will load your bundles from (optional)
+    publicPath: 'dist/'
+  }
+};
+```
+
+When using [Babel](http://babeljs.io/) you'll need to make sure that Babel can
+parse the dynamic import syntax but is not transforming it. For that you will need [babel-plugin-syntax-dynamic-import](https://yarnpkg.com/en/package/babel-plugin-syntax-dynamic-import).
+
+## Libraries
+
+### React Loadable
+
+[React Loadable](https://github.com/thejameskyle/react-loadable) provides you
+with a nice API for introducing code splitting into your app where-ever you
+render a component.
+
+**Before:**
+
+```js
+import OtherComponent from './OtherComponent';
+
+const MyComponent = () => (
+  <OtherComponent/>
+);
+```
+
+**After:**
+
+```js
+import Loadable from 'react-loadable';
+
+const LoadableOtherComponent = Loadable({
+  loader: () => import('./OtherComponent'),
+  loading: () => <div>Loading...</div>,
+});
+
+const MyComponent = () => (
+  <LoadableOtherComponent/>
+);
+```
+
+React Loadable helps you rendering
```

Comment:
I think there's a small typo here. Should it be "helps you render"/"helps with rendering"?

> React Loadable helps you rendering loading states, error states...